feat(react-scheduler): add CurrentTimeIndicator plugin (#2536)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/current-time-indicator.md

# CurrentTimeIndicator Plugin Reference

> A plugin that renders the current time indicator and shades that cover appointments and timetable cells up to the current time.

Reviewer: ```suggestion
A plugin that renders the current time indicator and the shading that covers appointments and timetable cells up to the current time.
```

## Import

Use the following statement to import the plugin with embedded theme components:

```js
import { CurrentTimeIndicator } from '@devexpress/dx-react-scheduler-material-ui';
```

You can import the themeless plugin if you want to use custom components:

```js
import { CurrentTimeIndicator } from '@devexpress/dx-react-scheduler';
```

## User Reference

### Dependencies

- [DayView](day-view.md) [Optional]
- [WeekView](week-view.md) [Optional]
- [MonthView](month-view.md) [Optional]
- [Appointments](appointments.md) [Optional]
- [DragDropProvider](drag-drop-provider.md) [Optional]

### Properties

Name | Type | Default | Description
-----|------|---------|------------
updateInterval | number | 60000 | An interval in milliseconds that specifies how frequently the indicator's position is updated.

> shadePastAppointments | boolean | `false` | Specifies whether past appointments should be shaded.

Reviewer 2: ```suggestion
shadePastAppointments | boolean | `false` | Specifies whether previous appointments should be shaded.
```

> shadePastCells | boolean | `false` | Specifies whether past cells should be shaded.

Reviewer: past appointments? past cells? Do you mean 'previous'?

Reviewer 2: We are describing a current time indicator, so past appointments are all appointments that ended before current time; similarly, past cells are timetable cells that represent time intervals that ended before current time.

Reviewer: Then you should use 'previous', since 'past' could indicate any previous time interval.

Reviewer 2: ```suggestion
shadePastCells | boolean | `false` | Specifies whether previous cells should be shaded.
```

indicatorComponent | ComponentType&lt;[CurrentTimeIndicator.IndicatorProps](#currenttimeindicatorindicatorprops)&gt; | | A component that renders the CurrentTimeIndicator.

## Interfaces

### CurrentTimeIndicator.IndicatorProps

Properties passed to the component that renders the CurrentTimeIndicator.

Field | Type | Description
------|------|------------

> top? | string | Specifies `top` CSS property in percentage of the containing element's height

Reviewer 2: ```suggestion
top? | string | Indicates the distance from the top edge of the containing element (usually, a timetable cell). The distance is measured as a percentage of the element's height.
```

> top? | string | Indicates the distance from the top edge of the containing element (usually, a timetable cell). The distance is measured as a percentage of the element's height.

Reviewer: ```suggestion
top? | string | Indicates the distance from the top edge of the containing element (usually, a timetable cell). The distance is a percentage of the element's height.
```

# CurrentTimeIndicator Plugin Reference

A plugin that renders the Scheduler's current time indicator

## Import

Use the following statement to import a plugin with embedded theme components:

```js
import { CurrentTimeIndicator } from '@devexpress/dx-react-scheduler-material-ui';
```

You can import the themeless plugin if you want to use custom components:

```js
import { CurrentTimeIndicator } from '@devexpress/dx-react-scheduler';
```

## User Reference

### Dependencies

- [DayView](day-view.md) [Optional]
- [WeekView](week-view.md) [Optional]
- [MonthView](month-view.md) [Optional]
- [Appointments](appointments.md) [Optional]
- [DragDropProvider](drag-drop-provider.md) [Optional]

### Properties

Name | Type | Default | Description
-----|------|---------|------------
updateInterval | number | 60000 | The time interval in milliseconds between Indicator's position updates.
reduceBrightnessOfPastAppointments | boolean | `false` | Specifies whether the brightness of past appointments should be reduced.

> shadePastCells | boolean | `false` | Specifies whether past cells should be shaded.

Reviewer 3: I think we should make these properties names similar. `shadePastCelss`, `shadePastAppointments` or something like that

indicatorComponent | ComponentType&lt;[CurrentTimeIndicator.IndicatorProps](#currenttimeindicatorindicatorprops)&gt; | | A component that renders the CurrentTimeIndicator.

## Interfaces

### CurrentTimeIndicator.IndicatorProps

Properties passed to the component that renders the CurrentTimeIndicator.

Field | Type | Description
------|------|------------

> currentTime | Date | Represents current time.

Reviewer 3: I don't understand why are we pass`currentTime` property instead of use Date.now()? It's not clear
